aidial_adapter_vertexai/llm/gemini_chat_completion_adapter.py

```diff
@@ -105,7 +118,36 @@ async def send_message_async(
                 tools=None,
             )
 
-            yield response.text
+            yield response
+
+    @staticmethod
+    async def process_chunks(
+        consumer: Consumer,
+        generator: Callable[[], AsyncIterator[GenerationResponse]],
+    ) -> AsyncIterator[str]:
+        completion = ""
+
+        async for chunk in generator():
+            if log.isEnabledFor(DEBUG):
+                log.debug(f"response chunk: {to_dict(chunk)}")
+
+            finish_reason = chunk.candidates[0].finish_reason
+
+            if (
+                completion == ""
+                and finish_reason == Candidate.FinishReason.OTHER
+            ):
+                # Misc finish reason could be usually fixed with a retry
+                raise RetryException(RuntimeError(UNEXPECTED_ERROR_MESSAGE))
```

Comment:
I think it's worth renaming the exception (e.g. to "class FinishReasonOtherError(RuntimeError)") and avoid nesting another one. The streaming status could be checked in the exception handler.

Reply:
Fixed